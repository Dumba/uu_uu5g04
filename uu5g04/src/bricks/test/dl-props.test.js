/**
 * Copyright (C) 2019 Unicorn a.s.
 * 
 * This program is free software; you can use it under the terms of the UAF Open License v01 or
 * any later version. The text of the license is available in the file LICENSE or at www.unicorn.com.
 * 
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
 * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See LICENSE for more details.
 * 
 * You may contact Unicorn a.s. at address: V Kapslovne 2767/2, Praha 3, Czech Republic or
 * at the email: info@unicorn.com.
 */

import React from 'react';
import {shallow} from 'enzyme';
import UU5 from "uu5g04";
import "uu5g04-bricks";
import enzymeToJson from 'enzyme-to-json';
import TestTools from "../../core/test/test-tools.js";


const TagName = "UU5.Bricks.Dl";

const CONFIG = {
  mixins: [
    "UU5.Common.BaseMixin",
    "UU5.Common.ElementaryMixin",
    "UU5.Common.SectionMixin",
    "UU5.Common.NestingLevelMixin",
    "UU5.Common.PureRenderMixin",
    "UU5.Common.ContentMixin",
    "UU5.Common.LevelMixin"
  ],
  props: {
    allowsTags: {
      values: ["Skoda.Account.Box"]
    }
  },
  requiredProps: {},
  opt: {
    shallowOpt: {
      disableLifecycleMethods: false
    },
    enzymeToJson: false
  }
};

describe(`${TagName}`, () => {
  TestTools.testProperties(TagName, CONFIG);
});

describe(`${TagName} example docKit`, () => {

  it(`${TagName} render without crash`, () => {
    const wrapper = shallow(
      <UU5.Bricks.Dl id={"uuID01"}>
        <UU5.Bricks.Dt id={"uuID02"} content='Google'/>
        <UU5.Bricks.Dd id={"uuID03"}
          content='It is an American multinational technology company specializing in Internet-related services and products that include online advertising technologies, search, cloud computing, software, and hardware.'
        />
      </UU5.Bricks.Dl>
    );
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });
});










