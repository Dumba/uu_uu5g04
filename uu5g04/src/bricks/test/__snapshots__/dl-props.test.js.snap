// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UU5.Bricks.Dl allowsTags="Skoda.Account.Box" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl checkGrammar=false 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl checkGrammar=true 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl checkHighlight=false 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl checkHighlight=true 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl checkSpaces=false 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl checkSpaces=true 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl className="my-classname" 1`] = `
<dl
  className="uu5-bricks-dl my-classname"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl content="This is Content from ContentMixin" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl content=12345 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl controlled=false 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl controlled=true 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl default props 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl disabled=false 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl disabled=true 1`] = `
<dl
  className="uu5-bricks-dl uu5-common-disabled"
  id="uuID"
  style={Object {}}
  title={null}
>
  <span
    className="uu5-common-disabled-cover"
    key="disabledCover"
  />
</dl>
`;

exports[`UU5.Bricks.Dl dynamic=false 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl dynamic=true 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl example docKit UU5.Bricks.Dl render without crash 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID01"
  style={Object {}}
  title={null}
>
  <Dt
    content="Google"
    id="uuID02"
    key="uuID02"
  />
  <Dd
    content="It is an American multinational technology company specializing in Internet-related services and products that include online advertising technologies, search, cloud computing, software, and hardware."
    id="uuID03"
    key="uuID03"
  />
</dl>
`;

exports[`UU5.Bricks.Dl footer={"type":"div","key":null,"ref":null,"props":{"children":"Footer of section"},"_owner":null,"_store":{}} 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
>
  <Footer
    content={
      <div>
        Footer of section
      </div>
    }
    id="uuID-footer"
    key="uuID-footer"
  />
</dl>
`;

exports[`UU5.Bricks.Dl header={"type":"div","key":null,"ref":null,"props":{"children":"Header of section"},"_owner":null,"_store":{}} 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
>
  <Header
    content={
      <div>
        Header of section
      </div>
    }
    id="uuID-header"
    key="uuID-header"
    underline={false}
  />
</dl>
`;

exports[`UU5.Bricks.Dl hidden=false 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl hidden=true 1`] = `
<dl
  className="uu5-bricks-dl uu5-common-hidden"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl id="myId" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="myId"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl ignoreHTML=false 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl ignoreHTML=true 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl level="0" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl level="1" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl level="2" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl level="3" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl level="4" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl level="5" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl level="6" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl level=0 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl level=1 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl level=2 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl level=3 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl level=4 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl level=5 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl level=6 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl mainAttrs=undefined 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl name="myName" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  name="myName"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl nestingLevel="bigBox" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl nestingLevel="bigBoxCollection" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl nestingLevel="box" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl nestingLevel="boxCollection" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl nestingLevel="container" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl nestingLevel="inline" 1`] = `
<dl
  className="uu5-bricks-dl uu-bricks-g4nixu"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl nestingLevel="page" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl nestingLevel="smallBox" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl nestingLevel="smallBoxCollection" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl nestingLevel="spa" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl noIndex=false 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl noIndex=true 1`] = `
<dl
  className="uu5-bricks-dl uu5-noindex"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl pureRender=false 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl pureRender=true 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl selected=false 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl selected=true 1`] = `
<dl
  className="uu5-bricks-dl uu5-common-selected"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl style="padding:8px; borderRadius: 2px" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={
    Object {
      "borderRadius": "2px",
      "padding": "8px",
    }
  }
  title={null}
/>
`;

exports[`UU5.Bricks.Dl style={"backgroundColor":"red","color":"blue"} 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={
    Object {
      "backgroundColor": "red",
      "color": "blue",
    }
  }
  title={null}
/>
`;

exports[`UU5.Bricks.Dl textCorrector=false 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl textCorrector=true 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl tooltip="My Tooltip" 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title="My Tooltip"
/>
`;

exports[`UU5.Bricks.Dl underline=false 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;

exports[`UU5.Bricks.Dl underline=true 1`] = `
<dl
  className="uu5-bricks-dl"
  id="uuID"
  style={Object {}}
  title={null}
/>
`;
